tests/ssl-opt.sh

```diff
@@ -14135,6 +14135,17 @@ run_test    "TLS 1.3: no HRR in case of PSK key exchange mode" \
             -c "Selected key exchange mode: psk$" \
             -c "HTTP/1.0 200 OK"
 
+# Legacy_compression_methods testing
+
+requires_gnutls
+requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_2
+requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_3
```

Comment:
```suggestion
requires_config_enabled MBEDTLS_SSL_SRV_C
requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_2
```
It makes sense to me to run it also when only TLS 1.2 is enabled.